traits/MLControl.php

```diff
@@ -241,6 +263,23 @@ public function getFallbackType()
         return defined('static::FALLBACK_TYPE') ? static::FALLBACK_TYPE : 'text';
     }
 
+    public function isFieldParentJsonable()
+    {
+        $names = HtmlHelper::nameToArray($this->formField->arrayName);
+        if (count($names) >= 2) {
+            // $names[0] is the Model, $names[1] is the top array name
+            $arrayName = $names[1];
+
+            if ($this->model->isClassExtendedWith('System\Behaviors\SettingsModel') ||
+                method_exists($this->model, 'isJsonable') && $this->model->isJsonable($arrayName)
+            )
+            {
+                return true;
+            }
+        }
+        return false;
+    }
```

Comment:
Could you provide some more context as to what / why these conditions make the field parent jsonable, especially the settings model check?